ocp_resources/utils.py

```diff
@@ -276,3 +276,39 @@ def _return_resource(_resource, _check_exists, _msg):
                     _check_exists=check_exists,
                     _msg=skip_create_warn_msg,
                 )
+
+
+def add_rules(role_object):
```

Comment:
I think this function should generate the rules and return then, not set them in the resource.
Should get `role_object.rules` and return the `_role_object.desired_state["rules"]`

Then in the resource you need to set it
`self.res["rules"] = _role_object.desired_state["rules"]`